app/Http/Requests/AddTagsRequest.php

```diff
@@ -16,7 +16,9 @@ public function rules()
         return [
             'photo_id' => 'required|exists:photos,id',
             'tags' => 'required|array',
-            'presence' => 'required|boolean'
+            'presence' => 'required|boolean',
+            'custom_tags' => 'nullable|array|max:3',
+            'custom_tags.*' => 'distinct:ignore_case|min:3|max:100'
```

Comment:
Whats with the 2 validation checks for custom_tags?

Reply:
The first one validates the custom_tags as a whole, so it should be an array with max 3 items. The second one validates every item in the array, they should be unique, min 3 characters and max 100 characters long.